docs/guide/modules/checkout.md

```diff
@@ -118,6 +118,68 @@ Component handling all the shipping logic
 - `notInMethods` - checks if passed method is present in `shippingMethods`
 
+## How to add a custom checkout step
+
+We now show an example of how to add a new step to the checkout page of Vue Storefront.
+
+The step is named `NewStep` and is placed just after the `PersonalDetails` step; changing the step's name and position requires small modifications to the procedure.
+
+### First, create the NewStep component
+
+1. **Create the NewStep component** according to your needs. To do it quickly, make a copy of the `PersonalDetails` component, name it `NewStep` and customize it.
+
+2. **Customize the sendDataToCheckout method** of the `NewStep` component so that it emits the event `checkout-after-newStep`; for example:
+```javascript
+    sendDataToCheckout () {
+      this.$bus.$emit('checkout-after-newStep', this.newStep, this.$v)
+    }
+```
+
+3. **Call the sendDataToCheckout method** when the button to the next section is clicked. This could be achieved in the template like this:
+```javascript
+    <button-full
+      @click.native="sendDataToCheckout"
+    >
+```
+
+
+### Then, modify the checkout component
+
+1. **Insert the NewStep component in the checkout template** at the desired position. For example, you could place it between the Personal Details and Shipping steps:
+```vue
+  <personal-details class="line relative" :is-active="activeSection.personalDetails" :focused-field="focusedField"/>
+  <new-step class="line relative" :is-active="activeSection.newStep">
+  <shipping class="line relative" :is-active="activeSection.shipping" v-if="!isVirtualCart"/>
+  <payment class="line relative" :is-active="activeSection.payment"/>
+  <order-review class="line relative" :is-active="activeSection.orderReview"/>
+```
+
+2. **Listen for the checkout-after-newStep event** by adding the following listener to the `beforeMount()` function:
+```javascript
+    this.$bus.$on('checkout-after-newStep', this.onAfterNewStep)
+```
+
+3. **Specify how to jump to the next step** by creating the method `onAfterNewStep`; in this example, the next step is the shipping form:
+```javascript
+    onAfterNewStep (receivedData, validationResult) {
+      this.newStep = receivedData
+      this.validationResults.newStep = validationResult
+      this.activateSection('shipping') // change 'shipping' to whatever you want the next step to be
+      this.saveNewStepData() // include only if newStep has state
+    }
+```
+Note that calling `activateSection('shipping')` is what ultimately shows the next checkout step to the user.
+
+4. **If needed, save NewStep state** by defining a non-empty method `saveNewStepData()`; for example:
+```javascript
+    saveNewStep () {
+      this.$store.dispatch('checkout/saveNewStep', this.newStep)
+    },
+```
+
+5. **Specify how to jump from the previous step to NewStep** by modifying the `onAfterPersonalDetails()` method, assuming `PersonalDetails` is the step that precedes our new step.
+
+
 ## Store
```

Comment:
Is this section waiting to be filled in or rather should be just removed?

Reply:
If you refer to the `Store` H2 section, that was there before I made my modifications... I can remove it in this pull request if needed.
PS: In the meanwhile I have improved the order of the steps of the howto.

Reply:
Yes, please remove it